### PR TITLE
Quality Assurance - QA Stages, Formatting, Guidelines and Web Quality Assurance Handbook

### PR DESCRIPTION
**Pull request by MaxHarrisonGit**
## Description
This PR Includes QA Stages, Formatting, Guidelines and Web Quality Assurance Handbook for Quality Assurance.
- QA Stages are at what point and stage should the Project be Tested.
- Formatting is how should the tests, results, and bugs be formatted and created.
- Guidelines is what needs to be followed either for QA or for Devs building with QA in mind.
- Web Quality Assurance Handbook is to outline and mark the Quality we expect for a Website that is entering QA.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds documentation)
- [ x] Breaking change (This is a significant change)

## Checklist
- [x ] I have ensured that I have used relative links where appropriate.
- [ x] I have proofread my documentation.

### CODE REVIEW

File: qualityassurance/webqualityassurancehandbook.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
+# 📗 Web Quality Assurance Handbook
+
+
```
Comment (AlexCatch) [inline qualityassurance/webqualityassurancehandbook.md]: Unnecessary whitespace

File: qualityassurance/webqualityassurancehandbook.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+Test extensively against assumptions in Jira, at the end of the day this is what the client is paying for. Below is a list of bullet points for common individual sections of a Laravel web app, although a lot of the core concepts apply regardless of web technology used (events bus, queues etc are all common design patterns)
+
+
```
Comment (AlexCatch) [inline qualityassurance/webqualityassurancehandbook.md]: Unnecessary whitespace

File: qualityassurance/webqualityassurancehandbook.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+It should be used as a guide to ensure a great & consistent experience for the end user.
+
+
```
Comment (AlexCatch) [inline qualityassurance/webqualityassurancehandbook.md]: Unnecessary whitespace

File: qualityassurance/webqualityassurancehandbook.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+-  Authentication when they are logged in
+-  Login & Logout works as expected
+-  Reset password works as expected
+
+
+#### Registration
+-  Email addresses are verified & unique (if applicable)
+-  Passwords meet minimum standards (min 8 chars)
+-  Passwords should not have maximum length restrictions (allow at least 255 characters).
+
```
Comment (AlexCatch) [inline qualityassurance/webqualityassurancehandbook.md]: IMO there should be only one newline between sections & not two.

File: qualityassurance/webqualityassurancehandbook.md [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+#### Permissions & Roles
+-  Ensure permissions & roles match up to possible actions (only admins can create users etc)
+
+
+### Whitebox
+
+#### General
+-  Filenames must follow standard coding conventions (UsersController.php)
```
Comment (AlexCatch) [inline qualityassurance/webqualityassurancehandbook.md]: There are two whitespaces between the `-` and the statement on each bulletpoint - this should probably be moved to just one

File: qualityassurance/qastagesformattingguidelines.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
+# ♻️ QA Testing Stages, Formatting and GuideLines
+
+This Section is to instruct on when to test, how you should be formatting the Bugs raised and GuideLines for QA
+
+## Developing with QA in mind
+
+### Commands
+When you are developing the project the project might require commands to run and work correctly, these can range from simple set up commands to automatic updating commands, a list of these commands and a breif description of what these commands do will help QA move more smoothly and avoid raising unnecessary questions.
```
Comment (AlexCatch) [inline qualityassurance/qastagesformattingguidelines.md]: brief is mispelt

File: qualityassurance/qastagesformattingguidelines.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+Some projects requires certain status or manipulation of the database to update systems and cause it to do the next stage/part of the system to test. We would like these to be formed into Commands, QA should not need to touch the database side to test the system, this leads to user error and this is leading it away from how the customer would user the system. So these processes need to put into commands where it will update the database for the required data with the correct process to ensure that it is changed to how the system will change it when on live.
```
[the remark this answers is not in the document]
Comment (AlexCatch) [reply qualityassurance/qastagesformattingguidelines.md]: I disagree with this line - Spending extra development time to write commands to automatically trigger stages outside of the lifecycle of the application I don't feel is effective use of dev time, QA should typically move through the application as a user so commands should usually never be necessary to trigger.

For testing that does require certain conditions to be met like a date being past a certain other date etc, changing the value in the database is easy enough & IMO doesn't warrant dedicated development time.

I do agree commands should be written for very complex processes & triggers that wouldn't be practical to achieve by the tester but these are far and few between, there are a lot more cost effective solutions though like manually triggering Chron jobs etc.

In relation to QA not having to touch the DB, it depends on the kind of testing taking place. black box testing needs access to the database. white box testing does not etc.

@MaxHarrisonGit I'd love to hear your opinions on this.

[the remark this answers is not in the document]
Comment (MaxHarrisonGit) [reply qualityassurance/qastagesformattingguidelines.md]: Thanks for the feedback, As we are doing black box and white box testing we would need the option to do it entirely from the front end with only the bare minimum manipulation of data to test, to ensure that everything it working when you don't manipulate the data. 
This was more in general, Because we want to do box White and Black box testing we need to have both taken account for when developing. 

In terms of doing simple commands these are "Required" however it would have QA if we have new users come in, seeing a large database and being told to just change 2 things in two what seems to be random tables, you can easily forget, so either having a nice simple command set there, or having documentation of what is exactly need to do and what the outcome is and why would would do it, either would help flow and help improve the speed of testing. 

@AlexCatch Would Like to hear your feedback to this 😄

File: qualityassurance/qastagesformattingguidelines.md [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+The structure of the tests are as follows:
+- Test Name
+- Priority
+    - Critical
+    - High
+    - Medium
+    - Low
+- Expected Outcome
+- Estimated time to complete testing
+- Area of Testing (Such as the what Pages/Screens)
```
Comment (dextermb) [inline qualityassurance/qastagesformattingguidelines.md]:
```suggestion
- Area of Testing (Such as the what pages/screens)
```

File: qualityassurance/qastagesformattingguidelines.md [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+
+## Test Cycle
+
+Test cycles are a collection of tests that need to be ran. Cycles should always be given a time frame and a time estimate to ensure that we are completing them in a timely manner and to enable management to work out work flows.
+
+### Tests
+For Stage 2, Stage 3 and stage 4, a list of tests is required, this is to keep track of what needs to be tested and to ensure that all functionality is tested and to ensure we don't do any unnecessary testing.
+The structure of the tests are as follows:
```
Comment (dextermb) [inline qualityassurance/qastagesformattingguidelines.md]:
```suggestion

The structure of the tests are as follows:
```

File: qualityassurance/qastagesformattingguidelines.md [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+- Test Name
+- Priority
+    - Critical
+    - High
+    - Medium
+    - Low
+- Expected Outcome
+- Estimated time to complete testing
+- Area of Testing (Such as the what Pages/Screens)
+- Test Data (If Required)
```
Comment (dextermb) [inline qualityassurance/qastagesformattingguidelines.md]:
```suggestion
 - Test Data (If required)
```

File: qualityassurance/qastagesformattingguidelines.md [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+
+### Testing Status
+Each test requires a status once completed. The Status are as follows;
```
Comment (dextermb) [inline qualityassurance/qastagesformattingguidelines.md]:
```suggestion
Each test requires a status once completed. The statuses are as follows:
```

File: qualityassurance/qastagesformattingguidelines.md [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+		- Test data used.
+		- Screenshots if relevant.
+		- Your Name.
+- Failed – The testing failed and is not working as expected.
+	- Data Required
+		- Notes.
+		- Test Data used.
+		- Bug Number Raised.
+		- Your Name.
+- Blocked – Testing has not been started as can’t be tested due to a Bug Already Raised by a failed test.
```
Comment (dextermb) [inline qualityassurance/qastagesformattingguidelines.md]:
```suggestion
- Blocked – Testing has not been started as can’t be tested due to a bug already raised by a failed test.
```

File: qualityassurance/qastagesformattingguidelines.md [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+	- Data Required
+		- Notes.
+		- Bug Number Raised.
+		- Your Name.
+- OOS (Out of Scope) – Test has been moved to OOS as it is not in the scope of testing for this Stage of Testing.
```
Comment (dextermb) [inline qualityassurance/qastagesformattingguidelines.md]:
```suggestion
- OOS (Out of Scope) – Test has been moved to OOS as it is not in the scope of testing for this stage of testing.
```

File: qualityassurance/qastagesformattingguidelines.md [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+
+### Raising A Bug
+
+When you find a bug, you will need to raise a bug on the Jira Service Desk.
+This bug will require multiple areas to be filled in, these are as Below:
+
+![Jira Bug Raising](images/jira/raising-bugs.PNG)
+
+### Test Data
+When testing on Staging and Prod we will require valid data and logons to access the system. These should be stored in a Google Docs that the Project Lead/Manager should have access to, These should be kept up to date. This should contain data that you need or is helpful to have when you are testing the given project.
```
Comment (dextermb) [inline qualityassurance/qastagesformattingguidelines.md]:
```suggestion
When testing on Staging and Prod we will require valid data and credentials. These should be stored in a Google Docs that the Project Lead/Manager should have access to, these should be kept up to date. This should contain data that you need, or is helpful, to have when you are testing the given project. 
```

File: qualityassurance/webqualityassurancehandbook.md [hunk 15]
```diff
@@ -0,0 +1,162 @@
+# 📗 Web Quality Assurance Handbook
+
+
+Test extensively against assumptions in Jira, at the end of the day this is what the client is paying for. Below is a list of bullet points for common individual sections of a Laravel web app, although a lot of the core concepts apply regardless of web technology used (events bus, queues etc are all common design patterns)
```
Comment (dextermb) [inline qualityassurance/webqualityassurancehandbook.md]:
```suggestion
Test extensively against assumptions in Jira, at the end of the day this is what the client is paying for. Below is a list of bullet points for common individual sections of a Laravel web app, although a lot of the core concepts apply regardless of web technology used (events bus, queues etc are all common design patterns).
```

File: qualityassurance/webqualityassurancehandbook.md [hunk 16]
```diff
@@ -0,0 +1,162 @@
...
+-  Icon is appropriate for the page
+
+
+#### Forms
+-  Invalid inputs correctly trigger validation errors
+-  All form inputs have associated labels
+-  Appropriate validation for different inputs (phone number, postcode)
+-  The forms are not too long & and are grouped logically
+-  The user is alerted of the form lifecycle (success, error)
+-  When modals are used in the context of updates, ensure inputs are correctly pre populated.
```
Comment (dextermb) [inline qualityassurance/webqualityassurancehandbook.md]:
```suggestion
-  When modals are used in the context of updates, ensure inputs are correctly pre populated
```

File: qualityassurance/webqualityassurancehandbook.md [hunk 17]
```diff
@@ -0,0 +1,162 @@
...
+-  Autofill is disabled on inputs where appropriate
+-  Inputs have correct type property ( emails have email type, phone numbers have tel type)
```
Comment (dextermb) [inline qualityassurance/webqualityassurancehandbook.md]:
```suggestion
-  Inputs have correct type property (emails have email type, phone numbers have tel type)
```

File: qualityassurance/webqualityassurancehandbook.md [hunk 18]
```diff
@@ -0,0 +1,162 @@
...
+-  Classes must follow standard coding conventions ( UsersController)
+-  Variables should follow camelCase & be descriptive (myName)
+-  All long running operations (at your discretion) are queued
+-  APIs follow REST architecture as closely as possible
+-  Auth middleware is setup & working correctly
+-  Event system is used for method side effects (Audit trail etc)
+-  Laravel Notifications are used to notify the user of system events (subscription expired, new friend request etc)
+-  Sensitive values are either hashed or encrypted (when required by law or common sense)
+-  Third party API integrations that can & will be re-used across projects should be implemented as external packages, assuming no existing open-source integration exists, to allow for independent testing and code reuse if appropriate.
+-  Ensure third party API keys (e.g. SES keys) are not present within .env.example and other relevant files outside of the production / development environments.
```
Comment (dextermb) [inline qualityassurance/webqualityassurancehandbook.md]:
```suggestion
-  Ensure third party API keys (e.g. SES keys) are not present within .env.example and other relevant files outside of the production / development environments
```

File: qualityassurance/webqualityassurancehandbook.md [hunk 19]
```diff
@@ -0,0 +1,162 @@
...
+-  If Travis CI is being used, ensure third party API keys are input directly into Travis CI’s website relevant to the project and make sure “Display value in build log” is not checked.
```
Comment (dextermb) [inline qualityassurance/webqualityassurancehandbook.md]:
```suggestion
-  If Travis CI is being used, ensure third party API keys are input directly into Travis CI’s website relevant to the project and make sure “Display value in build log” is not checked
```

File: qualityassurance/webqualityassurancehandbook.md [hunk 20]
```diff
@@ -0,0 +1,162 @@
...
+-  Inputs have correct type property ( emails have email type, phone numbers have tel type)
+
+
+#### Modals
+-  Modals are animated on both display and dismiss
+-  Modals are used where appropriate over pages
+
+
+#### Formatting
+-  Dates are formatted correctly for the appropriate audience.
```
Comment (dextermb) [inline qualityassurance/webqualityassurancehandbook.md]:
```suggestion
-  Dates are formatted correctly for the appropriate audience
```

File: qualityassurance/qastagesformattingguidelines.md [hunk 21]
```diff
@@ -0,0 +1,89 @@
...
+Some projects require certain status or manipulation of the database to update systems and cause it to do the next stage/part of the system to test. We would like these to be formed into commands, QA should not need to touch the database side to test the system, this leads to user error. This will allow the tester to carry out complex tasks without directly modify the database or sensitive areas of the application.
+
+
+### ENV
+Ensure that the 'env.example' is update and remove redundant fields/areas that are not required anymore.
+		
+## QA Stages
+For staging we are following a multi stage testing to enable us to ensure that the customer is getting a bug free and fully functioning system. This will be a merge between 'Agile' and 'Waterfall'.
+										
+### Stage 1 - Dev Testing
```
Comment (AlexCatch) [inline qualityassurance/qastagesformattingguidelines.md]:
```suggestion
### Stage 1 - Development Testing
```

File: qualityassurance/qastagesformattingguidelines.md [hunk 22]
```diff
@@ -0,0 +1,89 @@
...
+One check will be from a member of the development team to check over the code is to review the code itself. 
+The other check will be from QA, that will test the code changes itself, ensure that it is working and there are not bugs in the relevant area. This is more of a 'Spotlight Check', focused testing around each change that has been made and near areas that could be affected by this change. No matter how small an issue might seem it must be reported and signed off before continuing.
+Once both parties have accepted the code we will then go onto Stage 3.
+												
+### Stage 3 – M2S (Master to Staging) Testing
+This area of testing is large end to end testing. We start by creating a new user, with a cleared database (as much as can be cleared). Testing will be used as if the customer was using the system, this would include all different user types such as: client, admin, worker. If a customer has a web application, mobile application, and any other applications then this process will be repeated for each one until QA is confident to say that the applications are working as expected.
+
+### Stage 4 - Prod Testing
```
Comment (AlexCatch) [inline qualityassurance/qastagesformattingguidelines.md]:
```suggestion
### Stage 4 - Production Testing
```

File: qualityassurance/qastagesformattingguidelines.md [hunk 23]
```diff
@@ -0,0 +1,89 @@
...
+This area of testing is to test all key functionally of the application on prod to ensure that the application can function and does function as expected.
+If any urgent issues are found they should be raised and created as hotfixes, if they are not urgent they should go into the development branch and start the testing process again (from stage 2).
+
+#### Staging - Prod Merge Timing
```
Comment (AlexCatch) [inline qualityassurance/qastagesformattingguidelines.md]: I think this section might lean towards deployment procedures rather than QA & as merging into production isn't inherently dangerous at certain times unless deployment is paired with the merge as this could impair out ability to fight any fires.

File: qualityassurance/qastagesformattingguidelines.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (AlexCatch) [reply qualityassurance/qastagesformattingguidelines.md]: I'm happy to require documentation for procedures of steps and triggers that aren't obvious & commands for edge cases 👍

File: qualityassurance/qastagesformattingguidelines.md [hunk 24]
```diff
@@ -0,0 +1,89 @@
+# ♻️ QA Testing Stages, Formatting and GuideLines
+
+This Section is to instruct on when to test, how you should be formatting the bugs raised and guideLines for QA
```
Comment (ash123456789) [inline qualityassurance/qastagesformattingguidelines.md]:
```suggestion
This section is to instruct on when to test, how you should be formatting the bugs raised and guidelines for QA.
```

File: qualityassurance/qastagesformattingguidelines.md [hunk 25]
```diff
@@ -0,0 +1,89 @@
...
+
+## Developing with QA in mind
+### Commands
+When you are developing a project, the project might require commands to run and work correctly, these can range from simple set up commands to automatic updating commands, a list of these commands and a brief description of what these commands do will help QA move more smoothly and avoid raising unnecessary questions.
+
+Some projects require certain status or manipulation of the database to update systems and cause it to do the next stage/part of the system to test. We would like these to be formed into commands, QA should not need to touch the database side to test the system, this leads to user error. This will allow the tester to carry out complex tasks without directly modify the database or sensitive areas of the application.
```
Comment (ash123456789) [inline qualityassurance/qastagesformattingguidelines.md]:
```suggestion
Some projects require data to have certain statuses and manipulation of the database may be required to force the system onto the next process. We would like these to be formed into commands, QA should not need to touch the database side to test the system, this leads to user error. This will allow the tester to carry out complex tasks without directly modify the database or sensitive areas of the application.
```

File: qualityassurance/qastagesformattingguidelines.md [hunk 26]
```diff
@@ -0,0 +1,89 @@
...
+
+
+### ENV
+Ensure that the 'env.example' is update and remove redundant fields/areas that are not required anymore.
```
Comment (ash123456789) [inline qualityassurance/qastagesformattingguidelines.md]:
```suggestion
Ensure that the 'env.example' is up to date and remove redundant keys that are no longer required.
```

File: qualityassurance/qastagesformattingguidelines.md [hunk 27]
```diff
@@ -0,0 +1,89 @@
...
+		
+## QA Stages
+For staging we are following a multi stage testing to enable us to ensure that the customer is getting a bug free and fully functioning system. This will be a merge between 'Agile' and 'Waterfall'.
```
Comment (ash123456789) [inline qualityassurance/qastagesformattingguidelines.md]:
```suggestion
For staging, we are following a multi-stage testing methodology to enable us to ensure that the customer is getting a bug free and fully functioning system. This will be a merge between 'Agile' and 'Waterfall' methodologies.
```

File: qualityassurance/qastagesformattingguidelines.md [hunk 28]
```diff
@@ -0,0 +1,89 @@
...
+										
+### Stage 1 - Development Testing
+This stage is for the developers to test and ensure that there are not obvious bugs and that the code is working as expected before they create a PR (Pull Request). This will ensure that QA is not wasting time on obvious bugs that could have easily been spotted during development and be fix them before the pull request is made.
```
Comment (ash123456789) [inline qualityassurance/qastagesformattingguidelines.md]:
```suggestion
This stage is for the developers to test and ensure that there are no obvious bugs and that the code is working as expected before they create a PR (Pull Request). This will ensure that QA is not wasting time on obvious bugs that could have easily been spotted during development and be fixed them before the pull request is made.
```

File: qualityassurance/qastagesformattingguidelines.md [hunk 29]
```diff
@@ -0,0 +1,89 @@
...
+												
+### Stage 2 – PR (Pull Request) Testing
+At this stage the PR has been created and must adhere to the following criteria before being accepted.
+One check will be from a member of the development team to check over the code is to review the code itself.
```
Comment (ash123456789) [inline qualityassurance/qastagesformattingguidelines.md]:
```suggestion
One check will be from a member of the development team to check over the code and to review the code itself to ensure it meets our internal standards.
```

File: qualityassurance/qastagesformattingguidelines.md [hunk 30]
```diff
@@ -0,0 +1,89 @@
...
+										
+### Stage 1 - Development Testing
+This stage is for the developers to test and ensure that there are not obvious bugs and that the code is working as expected before they create a PR (Pull Request). This will ensure that QA is not wasting time on obvious bugs that could have easily been spotted during development and be fix them before the pull request is made.
+												
+### Stage 2 – PR (Pull Request) Testing
+At this stage the PR has been created and must adhere to the following criteria before being accepted.
+One check will be from a member of the development team to check over the code is to review the code itself. 
+The other check will be from QA, that will test the code changes itself, ensure that it is working and there are not bugs in the relevant area. This is more of a 'Spotlight Check', focused testing around each change that has been made and near areas that could be affected by this change. No matter how small an issue might seem it must be reported and signed off before continuing.
```
Comment (ash123456789) [inline qualityassurance/qastagesformattingguidelines.md]:
```suggestion
The other check will be from QA, that will test the code changes itself, ensure that it is working and there are no bugs in the relevant area(s). This is more of a 'Spotlight Check', focused testing around each change that has been made and near areas that could be affected by this change. No matter how small an issue might seem it must be reported and signed off before continuing.
```

File: qualityassurance/qastagesformattingguidelines.md [hunk 31]
```diff
@@ -0,0 +1,89 @@
...
+Once both parties have accepted the code we will then go onto Stage 3.
```
Comment (ash123456789) [inline qualityassurance/qastagesformattingguidelines.md]:
```suggestion
Once both parties have accepted the code we will then proceed to stage 3.
```

File: qualityassurance/qastagesformattingguidelines.md [hunk 32]
```diff
@@ -0,0 +1,89 @@
...
+												
+### Stage 3 – M2S (Master to Staging) Testing
+This area of testing is large end to end testing. We start by creating a new user, with a cleared database (as much as can be cleared). Testing will be used as if the customer was using the system, this would include all different user types such as: client, admin, worker. If a customer has a web application, mobile application, and any other applications then this process will be repeated for each one until QA is confident to say that the applications are working as expected.
```
Comment (ash123456789) [inline qualityassurance/qastagesformattingguidelines.md]:
```suggestion
This area of testing is large end to end testing. We start by creating a new user, with a cleared database (as much as can be cleared). Testing will be used as if the customer was using the system, this would include all different user types such as: client, admin, and/or worker. If a customer has a web application, mobile application, and any other application(s) then this process will be repeated for each application until QA is confident to say that the application(s) are working as expected.
```

File: qualityassurance/qastagesformattingguidelines.md [hunk 33]
```diff
@@ -0,0 +1,89 @@
...
+
+### Stage 4 - Production Testing
+This area of testing is to test all key functionally of the application on prod to ensure that the application can function and does function as expected.
+If any urgent issues are found they should be raised and created as hotfixes, if they are not urgent they should go into the development branch and start the testing process again (from stage 2).
+
+#### Staging - Prod Merge Timing
```
Comment (ash123456789) [inline qualityassurance/qastagesformattingguidelines.md]:
```suggestion
#### Staging - Production Merge Timing
```

File: qualityassurance/qastagesformattingguidelines.md [hunk 34]
```diff
@@ -0,0 +1,89 @@
...
+Timing is a very important point in the merge between Staging and Prod.
```
Comment (ash123456789) [inline qualityassurance/qastagesformattingguidelines.md]:
```suggestion
Timing is a very important point in the merge between staging and production.
```

File: qualityassurance/qastagesformattingguidelines.md [hunk 35]
```diff
@@ -0,0 +1,89 @@
...
+- We do not want to do it at peak working hours as this increases the chance of disrupting customer workflow.
```
Comment (ash123456789) [inline qualityassurance/qastagesformattingguidelines.md]:
```suggestion
- We do not want to merge at peak working hours as this increases the chance of disrupting customer workflow.
```

File: qualityassurance/qastagesformattingguidelines.md [hunk 36]
```diff
@@ -0,0 +1,89 @@
...
+- We do not want to do it towards the end of the day as if something were to go wrong it would delay resolution times until the next morning.
```
Comment (ash123456789) [inline qualityassurance/qastagesformattingguidelines.md]:
```suggestion
- We do not want to merge towards the end of the day as if something were to go wrong it would delay resolution times until the next morning.
```

File: qualityassurance/qastagesformattingguidelines.md [hunk 37]
```diff
@@ -0,0 +1,89 @@
...
+- We do not want to do it on Fridays as if something were to go wrong it would delay resolution times until the next working day.
```
Comment (ash123456789) [inline qualityassurance/qastagesformattingguidelines.md]:
```suggestion
- We do not want to merge on Fridays as if something were to go wrong it would delay resolution times until the next working day.
```

File: qualityassurance/qastagesformattingguidelines.md [hunk 38]
```diff
@@ -0,0 +1,89 @@
...
+It is worth noting that each customer has different situations and stances on when they want their applications updating. So it is always worth confirming with the customer before hand.
+
+## Test Cycle
+Test cycles are a collection of tests that need to be ran. Cycles should always be given a time frame and a time estimate to ensure that we are completing them in a timely manner and to enable management to organize work efficiently.
+
+### Tests
+For stage 2, stage 3 and stage 4, a list of tests is required, this is to keep track of what needs to be tested and to ensure that all functionality is tested and to ensure we don't do any unnecessary testing.
```
Comment (ash123456789) [inline qualityassurance/qastagesformattingguidelines.md]:
```suggestion
For stage 2, stage 3 and stage 4, a list of tests is required. This is to keep track of what needs to be tested and to ensure that all functionality is tested and to ensure we don't do any unnecessary testing.
```

File: qualityassurance/qastagesformattingguidelines.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (MaxHarrisonGit) [reply qualityassurance/qastagesformattingguidelines.md]: @AlexCatch I have edited to Document to Reflect this 😄Can give it a quick read over and confirm that We are on the same page

File: qualityassurance/qastagesformattingguidelines.md [hunk 39]
```diff
@@ -0,0 +1,89 @@
...
+		- Your name.
+
+### Raising A Bug
+When you find a bug, you will need to raise a bug on the 'Jira Service Desk'.
+This bug will require multiple areas to be filled in, these are as Below:
+
+![Jira Bug Raising](images/jira/raising-bugs.PNG)
+
+### Test Data
+When testing on 'Staging' and 'Prod' we will require valid data and credentials. These should be stored in a 'Google Docs' that the project lead/manager should have access to, these should be kept up to date. This should contain data that you need or is helpful to have when you are testing the given project.
```
Comment (ash123456789) [inline qualityassurance/qastagesformattingguidelines.md]:
```suggestion
When testing on `staging` and `production` we will require valid data and credentials. These should be stored in a 'Google Docs' document that the project lead/manager should have access to, these should be kept up to date. The document should contain data that you need or is helpful to have when you are testing the given project. 
```

File: qualityassurance/webqualityassurancehandbook.md [hunk 40]
```diff
@@ -0,0 +1,130 @@
+# 📗 Web Quality Assurance Handbook
+
+Test extensively against assumptions in Jira, at the end of the day this is what the client is paying for. Below is a list of bullet points for common individual sections of a Laravel web app, although a lot of the core concepts apply regardless of web technology used (events bus, queues etc are all common design patterns)
```
Comment (ash123456789) [inline qualityassurance/webqualityassurancehandbook.md]:
```suggestion
Test extensively against assumptions in Jira, at the end of the day this is what the client is paying for. Below is a list of bullet points for common individual sections of a Laravel web app, although a lot of the core concepts apply regardless of web technology used (events bus, queues etc. are all common design patterns)
```

File: qualityassurance/webqualityassurancehandbook.md [hunk 41]
```diff
@@ -0,0 +1,130 @@
...
+- Push notifications are received in a timely manner
+- They show correct & relevant information
+
+#### Permissions & Roles
+- Ensure permissions & roles match up to possible actions (only admins can create users etc)
+
+### Whitebox
+#### General
+- Filenames must follow standard coding conventions (UsersController.php)
+- Classes must follow standard coding conventions ( UsersController)
```
Comment (ash123456789) [inline qualityassurance/webqualityassurancehandbook.md]:
```suggestion
- Classes must follow standard coding conventions (UsersController)
```

**Comment by dextermb** [comment]
It's also worth noting that our git workflow says that commits should be commands: `updated -> update` 👍

**Comment by AlexCatch** [comment]
@wheatleyjj What's the next steps here are this includes the web quality assurance guide although that is currently open at #94 ?

**Comment by wheatleyjj** [comment]
@AlexCatch If this gets approved first then I presume this one goes in and my efforts are semi-wasted.

**Comment by dextermb** [comment]
> @AlexCatch If this gets approved first then I presume this one goes in and my efforts are semi-wasted.

🌈 Nothing is a waste

**Comment by AlexCatch** [comment]
@MaxHarrisonGit As we've made fair amount of changes in #94 IMO the web quality assurance guide should be removed from this PR & when the PR @wheatleyjj made has been merged in links can be updated etc.

What do you think about this?

**Comment by MaxHarrisonGit** [comment]
@AlexCatch I am fine with this, will remove it now 😄

**Comment by DivineOmega** [comment]
I believe we should get this both PR (#95) and @wheatleyjj's related PR (#94) merged in shortly, and resolve any conflicts as necessary.

**Comment by DivineOmega** [comment]
As has been discussed with a few people, this PR does currently contain a QA process different to how we currently work. Now this is outside of @MaxHarrisonGit's fork, feel free to make PRs that amend this.
